torch/fx/passes/utils/fuser_utils.py

```diff
@@ -50,19 +50,27 @@ def validate_partition(partition: NodeList) -> bool:
                 # external user node, need to expose as an output
                 outputs.append(user_node)
 
-    # perform DFS on the parition outputs
-    # if it reaches a node within the partition, then it found a cycle
+    # Set used to execlude nodes that have already been visited. 
+    # If a node has been visited, that node and all its children have
+    # been checked for cycles.
     visited: NodeSet = set()
```

Comment:
visited can be a local variable now, right? No reason for it to exist outside of the function, now?

Reply:
Well, it is reused for different outputs. If output `A` already check node `N`, we don't want to check `N` when processing output `B`.

Reply:
New commit pushed for improving clarify -- https://github.com/pytorch/pytorch/pull/95723/commits/2490f5f1bcd74598094322173300a395fd06cef1